### ISSUE TITLE
Running local functions fails without internet connection

### ISSUE THREAD

**Issue by milesfrain** [Azure/azure-functions-core-tools#2060]
I cannot run local Azure functions while offline.

Initial setup **with** internet connection
```
func init myapp --worker-runtime node --language javascript
cd myapp
npm install
func new --name HttpExample --template "HTTP trigger"
func host start
```

The above commands work fine. But then if I go offline and re-run `func host start`, I encounter this error:
```
Looking for extension bundle Microsoft.Azure.Functions.ExtensionBundle at /tmp/Functions/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle
[6/23/20 6:12:26 PM] Found a matching extension bundle at /tmp/Functions/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/1.1.1
[6/23/20 6:12:26 PM] Fetching information on versions of extension bundle Microsoft.Azure.Functions.ExtensionBundle available on https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/index.json
[6/23/20 6:12:26 PM] A host error has occurred during startup operation 'fd540883-3fd6-457a-a864-29232c8e703f'.
[6/23/20 6:12:26 PM] System.Net.Http: No such device or address. System.Private.CoreLib: No such device or address.
Value cannot be null.
Parameter name: provider
Application is shutting down...
```

#### Version Info
```
$ func --version 
2.7.2508
$ node --version
v10.21.0
$ lsb_release -d
Description:    Ubuntu 20.04 LTS
```

**Comment by anthonychu** [comment]
@soninaren Do you think we can skip this check if there's no internet connection?

**Comment by soninaren** [comment]
Yes, we should skip bundle load check when there is no internet.

**Comment by frarteaga** [comment]
> Yes, we should skip bundle load check when there is no internet.

When this change is available?

**Comment by fsudderthrenfroe** [comment]
Is this being worked on?

**Comment by akibrk** [comment]
Any update on this?

**Comment by iancoetzer** [comment]
Hi, I am having a similar issue while trying to debug offline. But my error message is slightly different:

[2023-04-19T16:34:23.264Z] A host error has occurred during startup operation '885dde36-fb96-4440-a98b-d4499664b3b4'.
[2023-04-19T16:34:23.272Z] System.Net.Http: No such host is known. (functionscdn.azureedge.net:443). System.Net.Sockets: No such host is known.
Value cannot be null. (Parameter 'provider')

**Comment by piotr-napadlek** [comment]
I understand that working on triggers like servicebus etc. would require an internet connection, but when I'm running azurite and want my function only to implement http endpoints and write to DevelopmentMode storage (also disabled appinsights logging), why would I require to be connected to internet? Any update on this?

**Comment by iancoetzer** [comment]
Hi, this is something that is also on my radar.

I do not myself understand why we must be connected to the internet to debug a local function which makes use of local storage?

**Comment by nawafmkutty** [comment]
Any updates on this issue?

The Url "functionscdn.azureedge.net" is unreachable even with the internet access.


[2023-08-14T08:44:14.742Z] Error building configuration in an external startup class.
[2023-08-14T08:44:14.743Z] Error building configuration in an external startup class. System.Net.Http: Network is unreachable (functionscdn.azureedge.net:443). System.Net.Sockets: Network is unreachable.
[2023-08-14T08:44:14.777Z] A host error has occurred during startup operation '7d6327e8-4f64-4706-b6c0-32c52a2fa6ab'.
[2023-08-14T08:44:14.778Z] Microsoft.Azure.WebJobs.Script: Error building configuration in an external startup class. System.Net.Http: Network is unreachable (functionscdn.azureedge.net:443). System.Net.Sockets: Network is unreachable.
Value cannot be null. (Parameter 'provider')
[2023-08-14T08:44:14.826Z] Host startup operation has been canceled

**Comment by offbeatful** [comment]
So for anyone who is looking for a solution to run their functions without Internet access:

I was required to remove `extensionBundle` section from `host.json` file. Why does this work?

It seems there is a code introduced in src/Azure.Functions.Cli/Actions/HostActions/Startup.cs that will check if `extensionBundle` is defined and if it is, it will set environment variables to check for latest version.

```csharp
private void SetBundlesEnvironmentVariables()
{
    var bundleId = ExtensionBundleHelper.GetExtensionBundleOptions(_hostOptions).Id;
    if (!string.IsNullOrEmpty(bundleId))
    {
        Environment.SetEnvironmentVariable("AzureFunctionsJobHost__extensionBundle__downloadPath", ExtensionBundleHelper.GetBundleDownloadPath(bundleId));
        Environment.SetEnvironmentVariable("AzureFunctionsJobHost__extensionBundle__ensureLatest", "true");
    }
}
```

That code was actually introduced as part of another issues:
```csharp
  // workaround for https://github.com/Azure/azure-functions-core-tools/issues/2097
  SetBundlesEnvironmentVariables();
```

I am running javascript functions and I confirmed that this section is not required for Development Model 4 when deploying in Azure. So `extensionBundle` section can be safely removed.

**Comment by KuSh** [comment]
> I am running javascript functions and I confirmed that this section is not required for Development Model 4 when deploying in Azure. So `extensionBundle` section can be safely removed.

Unfortunately, this workaround only works if you don't use extensions. I use durable functions and this problem is a real blocker

**Comment by perenna-dan** [comment]
I am too experiencing this issue. The point of local development should not require internet checks.

**Comment by liliankasem** [comment]
Issue should be fixed as part of the next core tools release (v4.8.0, eta for release is end of this week) - please open a new issue if you're on 4.8.0 and experiencing problems with this new feature - thanks!
